domain/fhir/zorginzage/service.go

```diff
@@ -0,0 +1,54 @@
+package zorginzage
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/monarko/fhirgo/STU3/datatypes"
+	"github.com/monarko/fhirgo/STU3/resources"
+
+	"github.com/nuts-foundation/nuts-demo-ehr/domain/fhir"
+)
+
+type Service interface {
+	CreateEpisode(ctx context.Context, dossierID, patientID string) (*fhir.EpisodeOfCare, error)
+}
+
+type service struct {
+	fhirClient fhir.Client
+}
+
+func NewService(fhirClient fhir.Client) Service {
+	return &service{
+		fhirClient: fhirClient,
+	}
+}
+
+func (service *service) CreateEpisode(ctx context.Context, dossierID, patientID string) (*fhir.EpisodeOfCare, error) {
+	episode := &fhir.EpisodeOfCare{
+		Base: resources.Base{
+			ID:           fhir.ToIDPtr(uuid.NewString()),
+			ResourceType: "EpisodeOfCare",
+		},
+		Type: []datatypes.CodeableConcept{
+			{
+				Coding: []datatypes.Coding{
+					{
+						System: &fhir.NutsCodingSystem,
+						Code:   fhir.ToCodePtr(dossierID),
```

Comment:
Types should be used to indicate the type of a resource. Like "etransfer" or "pregnancy", where the code is valid within the code system and the code is published.

Reply:
All right. I'll remove it but then we have to create a mapping in a SQL database as we can't search for episodes by our dossier ID